### PR TITLE
feat: Add game, custom bar animations + fix overflow behavior

### PR DESCRIPTION
**Pull request by kristofbolyai**

### CODE REVIEW

File: common/src/main/java/com/wynntils/core/features/overlays/BarOverlay.java [hunk 1]
```diff
@@ -31,11 +31,16 @@ public abstract class BarOverlay extends DynamicOverlay {
     @RegisterConfig("overlay.wynntils.barOverlay.flip")
     public final Config<Boolean> flip = new Config<>(false);
 
+    @RegisterConfig("overlay.wynntils.barOverlay.animated")
```
Comment (magicus) [inline common/src/main/java/com/wynntils/core/features/overlays/BarOverlay.java]: This is a bit odd config name and function. Maybe `animationTime`?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/core/features/overlays/BarOverlay.java]: Fixed.

File: common/src/main/java/com/wynntils/features/overlays/GameBarsOverlayFeature.java [hunk 2]
```diff
@@ -285,25 +272,99 @@ protected void renderBar(
...
+        protected abstract Texture getTexture();
+
+        protected abstract Texture getOverflowTexture();
+
+        protected abstract int getTextureY1();
+
+        protected abstract int getTextureY2();
+    }
+
+    protected static class HealthBarOverlay extends OverflowableBarOverlay {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/overlays/GameBarsOverlayFeature.java:304]: I am apparantly missing something about the game here. What does the "overflow" mean? How can the health (and other bars) overflow?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/overlays/GameBarsOverlayFeature.java:304]: Using certain abilities. You can double your mana/health.

Comment (magicus) [reply common/src/main/java/com/wynntils/features/overlays/GameBarsOverlayFeature.java:304]: And then we do not want to rescale the bar, but just change texture? Ok, I think I get this.

**Review by magicus — APPROVED** [review]

**Pull request merged**